### PR TITLE
first step of Action -> Object, SubAction -> Action

### LINKED ISSUE #179: Discussion: Syntax for SubAction
I find the current syntax for subaction non-composable. I think of subactions as just actions on an object.

**Describe the solution you'd like**
So, we can we define subaction using the `+` e.g.

`Action(....) + Action(....) ` so I am constantly doing more things to the object but I am just adding actions. Also, now actions are more composable, say if I wanna do the same subaction to two different objects I can just do `[Action(...), Action(...)] .+ Action(....)`

*Describe alternatives you've considered**
Hmm. None.

**Additional context**
Libraries in R like ggplot2 adopts this addition syntax to modify a plot. So I think it makes for Javis.jl too.

### LINKED ISSUE #204: Only one docstring for Action and SubAction
Currently it's quite hard to get an overview of all the different possibilities to define those two. Especially when having a look at 
```julia
julia>? SubAction
```

I think it makes more sense to define a docstring for one of them where we explain all the different functionalities.

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [ ] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [ ] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #179 
Closes #204 

**How did you address these issues with this PR? What methods did you use?**
Renaming from Action -> Object and SubAction -> Action
Adding an action to an object can be done with `+` (this will later be changed to `act!`)

```julia
using Javis

function ground(args...)
    background("black")
    sethue("white")
end

video = Video(800, 600)
javis(
    video,
    [
        BackgroundObject(1:100, ground),
        Object(1:100, (args...) -> circle(O, 50, :fill), Point(100, 10)) +
        Action(1:100, appear(:fade)),
    ];
    pathname = "current/object.gif",
)
```

### CODE REVIEW

**Comment by Wikunia** [comment]
You can now do your first review if you want @TheCedarPrince :smile: 
Things that I haven't done yet:
- Checked and changed docstrings
- used `act!` instead of `+`. Will do this when we move away from the array syntax inside Javis

**Comment by TheCedarPrince** [comment]
Sure thing! Would you like me to just ignore the things you have not changed?

**Comment by Wikunia** [comment]
Only ignore the docstrings for now. I'll add them later.
Would be good if you can:
- Check if your personal test code works as expected
- Check whether I missed some code changes

I'll change docstrings in this PR if that's okay and then merge it to v0.3
Afterwards I'll move over to the `act!` version with `render` instead of `javis`

**Comment by Wikunia** [comment]
I've added a docstring for `Action`. Would be great if you can review that @TheCedarPrince I'll change the style of the other docstrings to this if you think it makes sense.

File: src/Javis.jl [hunk 1]
```diff
@@ -270,121 +269,120 @@ function javis(
...
-        if rel_frame in get_frames(subaction)
-            subaction.func(video, action, subaction, rel_frame)
-            compute_transition!(subaction, video, rel_frame)
-            perform_transformation(subaction)
-        elseif rel_frame > last(get_frames(subaction))
-            # call the subaction on the last frame i.e. disappeared things stay disappeared
-            subaction.func(video, action, subaction, last(get_frames(subaction)))
+function draw_object(object, video, frame, origin_matrix)
+    # first compute and perform the global transformations of this object
+    translate(object.start_pos)
```
Comment (TheCedarPrince) [inline src/Javis.jl:329]: Is the preceding comment line still correct? All you are doing here it appears is translating the object to its start position on the frame.

File: src/Javis.jl [hunk 2]
```diff
@@ -415,13 +413,14 @@ for func in names(Luxor; imported = true)
 end
 
 export javis, latex
-export Video, Action, BackgroundAction, SubAction, Rel
+export Video, Object, BackgroundObject, Action, Rel
 export Line, Translation, Rotation, Transformation, Scaling
 export val, pos, ang, get_value, get_position, get_angle
-export projection, morph
+export projection, morph_to
```
Comment (TheCedarPrince) [inline src/Javis.jl:419]: Why the name change for morph?

File: src/luxor_overrides.jl [hunk 3]
```diff
@@ -221,12 +219,12 @@ animate text with [`appear`](@ref).
...
+Object(
     1:100,
     (args...) -> text("Hello Stream!"; halign = :center);
-    subactions = [
-        SubAction(1:15, sineio(), appear(:draw_text)),
-        SubAction(76:100, sineio(), disappear(:draw_text)),
+    actions = [
+        Action(1:15, sineio(), appear(:draw_text)),
+        Action(76:100, sineio(), disappear(:draw_text)),
     ],
```
Comment (TheCedarPrince) [inline src/luxor_overrides.jl:228]: Should we remove this syntax in favor of the `add!` syntax?

File: src/morphs.jl [hunk 4]
```diff
@@ -99,36 +98,41 @@ acirc(args...) = circle(Point(100,100), 50)
 
 video = Video(500, 500)
 javis(video, [
-    Action(1:100, ground),
-    Action(1:50, morph(astar, acirc)),
-    Action(51:100, morph(acirc, astar))
+    Object(1:100, ground),
+    Object(1:50, morph(astar, acirc)),
+    Object(51:100, morph(acirc, astar))
```
Comment (TheCedarPrince) [inline src/morphs.jl:103]: morph -> morph_to

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Just some minor suggested changes and questions. Otherwise, this looks fantastic and I am glad I was able to do some rigorous testing of it.

File: src/Javis.jl (hunk 1 above)
Comment (Wikunia) [reply src/Javis.jl:329]: The comment line should move down. Will fix

File: src/Javis.jl (hunk 2 above)
Comment (Wikunia) [reply src/Javis.jl:419]: The morph functionality changed into `Action` so it only has one argument now. Therefore it's not `morph(star, circle)` but `morph_to(circle)` if the parent object is a star. Hope that makes sense.

File: src/morphs.jl (hunk 4 above)
Comment (Wikunia) [reply src/morphs.jl:103]: Same deal. I changed the docs in #237

File: src/Javis.jl (hunk 2 above)
Comment (TheCedarPrince) [reply src/Javis.jl:419]: :+1:

**Review by TheCedarPrince — APPROVED** [review]

**Comment by TheCedarPrince** [comment]
LGTM - if you want to change that one other comment you mentioned here or in another PR works well either way. Feel free to merge when you are ready. Excited to have this syntax coming to be!

**Pull request merged**